packages/react/src/components/Toggle/next/Toggle.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright IBM Corp. 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { useControllableState } from '../../../internal/useControllableState';
+
+const { prefix } = settings;
+
+export function Toggle({
+  className,
+  defaultToggled = false,
+  disabled = false,
+  hideLabel = false,
+  id,
+  labelA = 'Off',
+  labelB = 'On',
+  labelText,
+  onClick = () => {},
+  onToggle,
+  size = 'md',
+  toggled,
+}) {
+  const [checked, setChecked] = useControllableState(
+    toggled,
+    onToggle,
+    defaultToggled
+  );
+
+  function handleClick(e) {
+    setChecked(!checked);
+    onClick(e);
+  }
+
+  const isSm = size === 'sm';
+  const sideLabel = hideLabel ? labelText : checked ? labelB : labelA;
+
+  const wrapperClasses = classNames(
+    `${prefix}--toggle`,
+    {
+      [`${prefix}--toggle--disabled`]: disabled,
+    },
+    className
+  );
+
+  const labelTextClasses = classNames(`${prefix}--toggle__label-text`, {
+    [`${prefix}--visually-hidden`]: hideLabel,
+  });
+
+  const appearanceClasses = classNames(`${prefix}--toggle__appearance`, {
+    [`${prefix}--toggle__appearance--sm`]: isSm,
+  });
+
+  const switchClasses = classNames(`${prefix}--toggle__switch`, {
+    [`${prefix}--toggle__switch--checked`]: checked,
+  });
+
+  return (
+    <div className={wrapperClasses}>
+      <button
+        id={id}
+        tabIndex={0}
+        className={`${prefix}--toggle__button`}
+        role="switch"
+        type="button"
+        aria-checked={checked}
+        disabled={disabled}
+        onClick={handleClick}
+      />
+      <label htmlFor={id} className={`${prefix}--toggle__label`}>
+        <span className={labelTextClasses}>{labelText}</span>
+        <div className={appearanceClasses}>
+          <div className={switchClasses}>
+            {isSm && (
+              <svg
+                className={`${prefix}--toggle__check`}
+                width="6px"
+                height="5px"
+                viewBox="0 0 6 5">
+                <path d="M2.2 2.7L5 0 6 1 2.2 5 0 2.7 1 1.5z" />
+              </svg>
+            )}
+          </div>
+          <span className={`${prefix}--toggle__text`} aria-hidden="true">
+            {sideLabel}
```

Comment:
Should the label be hidden with `aria-hidden`? I might be misunderstanding the intent of `sideLabel` so just wanted to check

Reply:
The `sideLabel` is only meant to convey whether the control is on or off which screen readers already read out based on the `aria-checked` attribute. That means, if the `sideLabel` is not hidden from the accessiblity tree, screen readers are picking up "{Label} Off, off, switch" instead of "{Label}, off, switch".

![image](https://user-images.githubusercontent.com/28265588/116055595-26fa6900-a67d-11eb-9f77-1ed6c2af6b8c.png)

Even though the `labelText` is rendered at the side when `hideLabel` is `true`, the actual label text is just visually hidden and therefore still picked up.